### PR TITLE
dac: ltc2672: Remove float type usage and other fixes

### PR DESCRIPTION
**Pull request by SaikiranGudla**
Remove float type usage, computations and replace them with uint32_t type.
Add range checks as required.
Remove incorrect uint8_t type casting in ltc2672
transaction function definition.

## Pull Request Description

Please replace this with a detailed description and motivation of the changes. 
You can tick the checkboxes below with an 'x' between square brackets or just check them after publishing the PR. 
If this PR contains a breaking change, list dependent PRs and try to push all related PRs at the same time.

## PR Type
- [x] Bug fix (change that fixes an issue)
- [ ] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

**Comment by buha** [comment]
"Remove incorrect uint8_t type casting in ltc2672
transaction function definition."

This should be a separate commit, please split it.

**Comment by buha** [comment]
Range checks could also be a separate commit, so you'd have 3 in total.

File: drivers/dac/ltc2672/ltc2672.h [hunk 1]
```diff
@@ -68,9 +68,10 @@
 #define LTC2672_MAX_CONFIG_MASK  		15
 #define LTC2672_MAX_TOGGLE_MASK  		31
 #define LTC2672_BIT_SHIFT_12BIT			4
-#define LTC2672_NUM_MUX_SELCETS			22
+#define LTC2672_NUM_MUX_SELECTS			22
 #define LTC2672_NUM_CURRENT_SPANS		10
 #define LTC2672_NUM_FAULTS				7
+#define MICROAMP_PER_MILLIAMP			1000
```
Comment (amiclaus) [inline drivers/dac/ltc2672/ltc2672.h:74]: this should go in `no_os_units.h` in a separate commit, prior to your driver patch.
https://github.com/analogdevicesinc/no-OS/blob/0cd0ac78cd95f13a14ae32a2daeb430c0444ba3b/include/no_os_units.h#L68-L70

Comment (SaikiranGudla) [reply drivers/dac/ltc2672/ltc2672.h:74]: Shall I open a separate PR for this and then use the no_os_util for this PR?

Comment (amiclaus) [reply drivers/dac/ltc2672/ltc2672.h:74]: You can do a separate commit in this PR. Prior to the commits for ltc2672

**Comment by SaikiranGudla** [comment]
Changelog:

Broke down single commit into 4 separate commits for each fix/update.

**Review by buha — APPROVED** [review]

**Pull request merged**